### ISSUE TITLE
[Travis CI] Test broken size fails with GCC 6.5

### ISSUE THREAD

**Issue by Farwaykorse** [Microsoft/GSL#745]
The span.size() bug reported in #590 remains in GCC 6.5 (as now used in Travis CI).
The test added with #704 now breaks the CI: [GSL build report](https://travis-ci.org/Microsoft/GSL/jobs/454937553)

According to the [gcc bugtracker: 83116](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=83116), the fix was supposed to have landed in GCC 6.5.

I [added a fix](https://github.com/Microsoft/GSL/pull/741/commits/84f57949abd4b69ac4003daa16537202096ea71f) while working on #741, should I make it a separate pull request?

**Comment by beinhaerter** [comment]
Thanks for opening the issue. I just had the same problem with PR #748.

**Comment by annagrin** [comment]
@Farwaykorse it would be great if you added a separate pull request for the span.size() bug test. Thanks!

**Comment by annagrin** [comment]
@Farwaykorse nevermind, I will copy your fix to unblock the builds, thanks!
